jsapp/js/components/processing/singleProcessingHeader.tsx

```diff
@@ -116,20 +124,36 @@ class SingleProcessingHeader extends React.Component<SingleProcessingHeaderProps
 
   /** Goes back to Data Table route for given project. */
   onDone() {
-    // If there are any changes to the data, we need to ensure that the latest
-    // asset is available in the Data Table, when it will rebuild itself, so
-    // that all the columns are rendered. This is needed for the case when user
-    // added/deleted transcript or translation, as editing the text value for it
-    // is already handled properly by Data Table code.
+    // HACK: If there are any changes to the data, we need to ensure that
+    // the latest asset is available in the Data Table, when it will rebuild
+    // itself, so that all the columns are rendered. This is needed for the case
+    // when user added/deleted transcript or translation (editing the text
+    // value for it is already handled properly by Data Table code).
     if (!singleProcessingStore.isPristine) {
-      // Here we delete the cached asset object from the cache, so that when
-      // `FormSubScreens` (a parent of Data Table) starts loading in a moment,
-      // it would fetch latest asset and make Data Table use it.
+      // Mark button as pending to let user know we wait for stuff.
+      this.setState({isDoneButtonPending: true});
+
+      // We don't need to add these listeners prior to this moment, and we don't
+      // need to cancel them, as regardless of outcome, we will navigate out of
+      // current view.
+      actions.resources.loadAsset.completed.listen(
+        this.navigateToDataTable.bind(this)
+      );
 
-      // JESSICA TODO: here add a line that removes the asset
-    }
+      // For failed load we still navigate to Data Table, as this is not
+      // something that would cause a massive disruption or data loss
+      actions.resources.loadAsset.failed.listen(
+        this.navigateToDataTable.bind(this)
+      );
```

Comment:
Does this need to get added to `this.unlisteners`? If the component unmounts before `loadAsset.completed`/`loadAsset.failed` fires, would this cause a memory leak?

Reply:
Right, better be safe here :) Added both to unlisteners